{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "### Neovim version (nvim -v)\n\nv0.7.2\n\n### Vim (not Nvim) behaves the same?\n\nno, vim doesn't have checkhealth\n\n### Operating system/version\n\nWindows 10.0.19044\n\n### Terminal name/version\n\nWindows Terminal 1.14.1962.0\n\n### $TERM environment variable\n\nunset\n\n### Installation\n\nextracted Github release build and set $PATH\n\n### How to reproduce the issue\n\nInstall [pyenv-win](https://github.com/pyenv-win/pyenv-win) and download some python enviroment. Then launch neovim and run `:checkhealth`\n\n### Expected behavior\n\nIt doesn't instantly fail\n\n### Actual behavior\n\nIt fails with the exception:\r\n```\r\nprovider: health#provider#check\r\n 13 ========================================================================\r\n 14   - ERROR: Failed to run healthcheck for \"provider\" plugin. Exception:\r\n 15     function health#check[20]..health#provider#check[2]..<SNR>47_check_python[13]..<SNR>47_check_for_pyenv[12]..            <SNR>47_system, line 11\r\n 16     Vim(let):E903: Process failed to start: no such file or directory: \"C:\\Users\\Ross\\.pyenv\\pyenv-win\\bin\\pyenv\"\r\n ```\r\n Indicating that the exception occurs on this line:\r\n https://github.com/neovim/neovim/blob/c6181a672a2fdb2cb89ddf85b5d6ab15b10a0996/runtime/autoload/health/provider.vim#L298\r\n Even though that directory does exist on my system. I also have python3_host_prog set to a virtual environment, but whether it is set or not it has the exception as it doesn't even reach checking if it is set.\r\n \r\n I have fixed it on my computer though by applying the following patch:\r\n ```patch\r\ndiff --git \"a/nvim\\share\\nvim\\runtime\\autoload\\health\\provider.vim\" \"b/nvim\\share\\nvim\\runtime\\autoload\\health\\provider.vim\"\r\nindex a01cb96..3c353b9 100644\r\n--- \"a/nvim\\share\\nvim\\runtime\\autoload\\health\\provider.vim\"\r\n+++ \"b/nvim\\share\\nvim\\runtime\\autoload\\health\\provider.vim\"\r\n@@ -295,7 +295,7 @@ function! s:check_python() abort\r\n     return\r\n   endif\r\n \r\n-  let [pyenv, pyenv_root] = s:check_for_pyenv()\r\n+  let [pyenv, pyenv_root] = ['', ''] \r\n \r\n   if exists('g:'.host_prog_var)\r\n     call health#report_info(printf('Using: g:%s = \"%s\"', host_prog_var, get(g:, host_prog_var)))\r\n\r\n```\r\nOnce applied the provider health check passes:\r\n```\r\n17 ## Python 3 provider (optional)\r\n 18   - INFO: Using: g:python3_host_prog = \"C:\\dev\\nvim\\py3nvim\\Scripts\\python.exe\"\r\n 19   - INFO: Executable: C:\\dev\\nvim\\py3nvim\\Scripts\\python.exe\r\n 20   - INFO: Python version: 3.10.0^M\r\n 21   - INFO: pynvim version: 0.4.3^M\r\n 22   - OK: Latest pynvim is installed.\r\n ```\r\n Looking through some PRs, it looks like #12437 may solve this, but has been stalled. ",
    "closed_at": null,
    "comment_data": [
        {
            "author_association": "NONE",
            "body": "Both a report and information for any other people who see this happen.",
            "created_at": "2022-08-16T01:22:29Z",
            "html_url": "https://github.com/neovim/neovim/issues/19788#issuecomment-1216039595",
            "id": 1216039595,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19788",
            "node_id": "IC_kwDOAPphoM5Ie0qr",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1216039595/reactions"
            },
            "updated_at": "2022-08-16T01:22:29Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1216039595",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/18294397?v=4",
                "events_url": "https://api.github.com/users/RossSmyth/events{/privacy}",
                "followers_url": "https://api.github.com/users/RossSmyth/followers",
                "following_url": "https://api.github.com/users/RossSmyth/following{/other_user}",
                "gists_url": "https://api.github.com/users/RossSmyth/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/RossSmyth",
                "id": 18294397,
                "login": "RossSmyth",
                "node_id": "MDQ6VXNlcjE4Mjk0Mzk3",
                "organizations_url": "https://api.github.com/users/RossSmyth/orgs",
                "received_events_url": "https://api.github.com/users/RossSmyth/received_events",
                "repos_url": "https://api.github.com/users/RossSmyth/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/RossSmyth/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/RossSmyth/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/RossSmyth"
            }
        },
        {
            "author_association": "CONTRIBUTOR",
            "body": "Please test https://github.com/neovim/neovim/pull/12437.\r\nIf it fixes the problem, it should be merged.",
            "created_at": "2022-08-16T01:26:49Z",
            "html_url": "https://github.com/neovim/neovim/issues/19788#issuecomment-1216041570",
            "id": 1216041570,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19788",
            "node_id": "IC_kwDOAPphoM5Ie1Ji",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1216041570/reactions"
            },
            "updated_at": "2022-08-16T01:26:49Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1216041570",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/41495?v=4",
                "events_url": "https://api.github.com/users/Shougo/events{/privacy}",
                "followers_url": "https://api.github.com/users/Shougo/followers",
                "following_url": "https://api.github.com/users/Shougo/following{/other_user}",
                "gists_url": "https://api.github.com/users/Shougo/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/Shougo",
                "id": 41495,
                "login": "Shougo",
                "node_id": "MDQ6VXNlcjQxNDk1",
                "organizations_url": "https://api.github.com/users/Shougo/orgs",
                "received_events_url": "https://api.github.com/users/Shougo/received_events",
                "repos_url": "https://api.github.com/users/Shougo/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/Shougo/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/Shougo/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/Shougo"
            }
        },
        {
            "author_association": "NONE",
            "body": "I have hit this again reinstalling on a new computer.\r\n\r\nThe pull mentioned may or may not work, but rebasing is not something I want to do. My patch still works though and so that all I did.",
            "created_at": "2023-01-07T06:37:51Z",
            "html_url": "https://github.com/neovim/neovim/issues/19788#issuecomment-1374392645",
            "id": 1374392645,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19788",
            "node_id": "IC_kwDOAPphoM5R65FF",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1374392645/reactions"
            },
            "updated_at": "2023-01-07T06:37:51Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1374392645",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/18294397?v=4",
                "events_url": "https://api.github.com/users/RossSmyth/events{/privacy}",
                "followers_url": "https://api.github.com/users/RossSmyth/followers",
                "following_url": "https://api.github.com/users/RossSmyth/following{/other_user}",
                "gists_url": "https://api.github.com/users/RossSmyth/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/RossSmyth",
                "id": 18294397,
                "login": "RossSmyth",
                "node_id": "MDQ6VXNlcjE4Mjk0Mzk3",
                "organizations_url": "https://api.github.com/users/RossSmyth/orgs",
                "received_events_url": "https://api.github.com/users/RossSmyth/received_events",
                "repos_url": "https://api.github.com/users/RossSmyth/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/RossSmyth/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/RossSmyth/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/RossSmyth"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "The providers/healtchecks were rewritten. Can you try the development version? \r\n\r\n- The [Releases](https://github.com/neovim/neovim/releases/tag/nightly) page has pre-built archives for Linux/Windows/macOS. \r\n- Check `:version` to confirm that you are using the latest version.",
            "created_at": "2024-02-15T12:54:19Z",
            "html_url": "https://github.com/neovim/neovim/issues/19788#issuecomment-1946043443",
            "id": 1946043443,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19788",
            "node_id": "IC_kwDOAPphoM5z_kQz",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1946043443/reactions"
            },
            "updated_at": "2024-02-15T12:54:19Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1946043443",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1359421?v=4",
                "events_url": "https://api.github.com/users/justinmk/events{/privacy}",
                "followers_url": "https://api.github.com/users/justinmk/followers",
                "following_url": "https://api.github.com/users/justinmk/following{/other_user}",
                "gists_url": "https://api.github.com/users/justinmk/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/justinmk",
                "id": 1359421,
                "login": "justinmk",
                "node_id": "MDQ6VXNlcjEzNTk0MjE=",
                "organizations_url": "https://api.github.com/users/justinmk/orgs",
                "received_events_url": "https://api.github.com/users/justinmk/received_events",
                "repos_url": "https://api.github.com/users/justinmk/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/justinmk/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/justinmk/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/justinmk"
            }
        }
    ],
    "comments": 4,
    "comments_url": "https://api.github.com/repos/neovim/neovim/issues/19788/comments",
    "created_at": "2022-08-16T01:19:20Z",
    "events_url": "https://api.github.com/repos/neovim/neovim/issues/19788/events",
    "html_url": "https://github.com/neovim/neovim/issues/19788",
    "id": 1339692425,
    "labels": [
        {
            "color": "f9d0c4",
            "default": true,
            "description": "issues reporting wrong behavior",
            "id": 77997474,
            "name": "bug",
            "node_id": "MDU6TGFiZWw3Nzk5NzQ3NA==",
            "url": "https://api.github.com/repos/neovim/neovim/labels/bug"
        },
        {
            "color": "c5def5",
            "default": false,
            "description": null,
            "id": 144672485,
            "name": "provider",
            "node_id": "MDU6TGFiZWwxNDQ2NzI0ODU=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/provider"
        },
        {
            "color": "FBCA04",
            "default": false,
            "description": "waiting for reply from the author",
            "id": 248229439,
            "name": "needs:response",
            "node_id": "MDU6TGFiZWwyNDgyMjk0Mzk=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/needs:response"
        }
    ],
    "labels_url": "https://api.github.com/repos/neovim/neovim/issues/19788/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "I_kwDOAPphoM5P2hWJ",
    "number": 19788,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/neovim/neovim/issues/19788/reactions"
    },
    "repository_url": "https://api.github.com/repos/neovim/neovim",
    "state": "open",
    "state_reason": null,
    "timeline_url": "https://api.github.com/repos/neovim/neovim/issues/19788/timeline",
    "title": "Windows python provider check failure",
    "updated_at": "2024-02-15T12:54:25Z",
    "url": "https://api.github.com/repos/neovim/neovim/issues/19788",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/18294397?v=4",
        "events_url": "https://api.github.com/users/RossSmyth/events{/privacy}",
        "followers_url": "https://api.github.com/users/RossSmyth/followers",
        "following_url": "https://api.github.com/users/RossSmyth/following{/other_user}",
        "gists_url": "https://api.github.com/users/RossSmyth/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/RossSmyth",
        "id": 18294397,
        "login": "RossSmyth",
        "node_id": "MDQ6VXNlcjE4Mjk0Mzk3",
        "organizations_url": "https://api.github.com/users/RossSmyth/orgs",
        "received_events_url": "https://api.github.com/users/RossSmyth/received_events",
        "repos_url": "https://api.github.com/users/RossSmyth/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/RossSmyth/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/RossSmyth/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/RossSmyth"
    }
}